internal/storage/image.go

```diff
@@ -249,13 +255,14 @@ func (svc *imageService) buildImageCacheItem(systemContext *types.SystemContext,
 			return imageCacheItem{}, err
 		}
 	}
-
+	name, _, _ := sortNamesByType(image.Names)
```

Comment:
How is this supposed to work with images that have multiple names?

Reply:
Are you asking how `sortNamesByType` will behave with multiple names?

Reply:
I did read that code. I don’t see that the current implementation provides a predictable behavior for users (the order of `image.Names`, in the current implementation, seems to depend on the order in which image names were pulled, and that’s _not at all promised by the API_).

My first guess is that all names in `image.Names` should be processed by the filter, but I didn’t look into the use cases of pinned images.

Reply:
> My first guess is that all names in image.Names should be processed by the filter, but I didn’t look into the use cases of pinned images.

the idea is to disqualify an image for kubelet garbage collection. Since the order of the names isn't guaranteed, it does sound like each should be considered to see if the image should be disqualified